packages/store/addon/-private/system/store.js

```diff
@@ -2126,15 +2133,26 @@ const Store = Service.extend({
       let adapter = this.adapterFor(internalModel.modelName);
       let operation;
 
-      if (internalModel.currentState.stateName === 'root.deleted.saved') {
-        resolver.resolve();
-        continue;
-      } else if (internalModel.isNew()) {
-        operation = 'createRecord';
-      } else if (internalModel.isDeleted()) {
-        operation = 'deleteRecord';
+      if (!ENABLE_RECORD_DATA_STATE) {
+        if (internalModel.currentState.stateName === 'root.deleted.saved') {
```

Comment:
where does the handling of this state move?

Reply:
https://github.com/emberjs/data/pull/6204/files#diff-1080751ea355f7ab0c457d1785c50695R2103